packages/modules/event-bus-redis/src/services/event-bus-redis.ts

```diff
@@ -71,78 +81,114 @@ export default class RedisEventBusService extends AbstractEventBusModuleService
     },
   }
 
-  /**
-   * Emit a single event
-   * @param {string} eventName - the name of the event to be process.
-   * @param data - the data to send to the subscriber.
-   * @param options - options to add the job with
-   */
-  async emit<T>(
-    eventName: string,
-    data: T,
-    options: Record<string, unknown>
-  ): Promise<void>
+  private buildEvents<T>(
+    eventsData: Message<T>[],
+    options: BulkJobOptions = {}
+  ): IORedisEventType<T>[] {
+    const opts = {
+      // default options
+      removeOnComplete: true,
+      attempts: 1,
+      // global options
+      ...(this.moduleOptions_.jobOptions ?? {}),
+      ...options,
+    }
+
+    return eventsData.map((eventData) => {
+      return {
+        name: eventData.eventName,
+        data: {
+          eventName: eventData.eventName,
+          data: eventData.body.data,
+        },
```

Comment:
I actually noticed this too but I thought it was intentional. The case is the same for the local event bus.

Reply:
I don't think the metadata should be removed as it is metadata information that we send to the user, but it could also be a place where the user add some metadata such as timestamp of any other information, it could be open ended